lib/msf/core/module_manager.rb

```diff
@@ -96,7 +96,7 @@ def create(name, aliased_as: nil)
             module_instance = create(aliased, aliased_as: "#{type}/#{name}")
           else
             module_reference_name = names.join("/")
-            module_instance = set.create(module_reference_name)
+            module_instance = set.has_key?(module_reference_name) ? set.create(module_reference_name) : nil
```

Comment:
The function that takes ~1 second to execute is the PayloadSet's `recalculate` method. There's a comment on the code that this approach may not work as we expect:

https://github.com/rapid7/metasploit-framework/blob/4c06146900360f6cd870e69040722d97cfac4a29/lib/msf/core/payload_set.rb#L7-L17

We might need a different approach to the above, but it's worth double checking.

Reply:
Poking further at this; I'm concerned that this approach breaks the law of demeter, and may not work well with some of the assumptions that are already in place within framework. In particular it seems like module set already has some nil handling in place:

https://github.com/rapid7/metasploit-framework/blob/4c06146900360f6cd870e69040722d97cfac4a29/lib/msf/core/module_set.rb#L42-L50

Which is actually why the slow `recalculate` method gets called in the first place. It might be worth doing a bit of git fu to learn why that logic was added in the first place, to make sure this approach doesn't break existing logic etc.

Reply:
Cross referencing some discussion on bootup/missing command performance - https://github.com/rapid7/metasploit-framework/pull/7292#discussion_r399343619